src/java.base/share/classes/com/sun/crypto/provider/Poly1305.java

```diff
@@ -165,11 +173,39 @@ void engineUpdate(byte[] input, int offset, int len) {
                 blockOffset = 0;
             }
         }
-        while (len >= BLOCK_LENGTH) {
-            processBlock(input, offset, BLOCK_LENGTH);
-            offset += BLOCK_LENGTH;
-            len -= BLOCK_LENGTH;
+
+        int blockLength = (len/BLOCK_LENGTH) * BLOCK_LENGTH;
+        byte[] aBytes = this.a.asByteArray(BLOCK_LENGTH+1);
+        byte[] rBytes = this.r.asByteArray(BLOCK_LENGTH);
+
+        byte[] exp = null;
+        { // DELETE: Fuzzing
+            int offsetFuzz = offset;
+            int lenFuzz = len;
+            while (lenFuzz >= BLOCK_LENGTH) {
+                // FIXME: check this.*
```

Comment:
Should this fixme note be removed or what exactly needs fixing here?

Reply:
I forgot to remove FIXME during cleanup. I believe I was leaving a note for myself to see if there were any off-by-one errors in the loop, but I checked all that with stepping with a debugger..

(Yes, it should be deleted. Well.. this whole block, from line 182)